### PR TITLE
Ability to control illuminators and hex string packing.

### PR DESCRIPTION
**Pull request by Leo-MathGuy**
Mindustry uses `control color link a_color 0 0 0` to control illuminatior colors.

Here, 2 features are added:

Hex variable detection+packing:
`red = "%ff0000"` -> `set red %ff0000` (hex detection added as no quotes for hex)
`half_transparent_white = "%ffffff88` -> `set half_transparent_white %ffffff88` 

Illuminator control:
```
red = "%ff0000"
illuminator1.color(red)
```
To
```
set red %ff0000
control color illuminator1 red 0 0 0
```

### CODE REVIEW

**Comment by Leo-MathGuy** [comment]
EDIT: Now added Screen.col(color)

Uses a packed color variable to set color. Screen.color() but with packed variables. Mindustry has an option for that.

File: pyndustri.pyi [hunk 1]
```diff
@@ -347,6 +347,8 @@ class Controllable(ABC):
         """Sets the link to shoot or not, and if shooting, the position."""
     def ceasefire(self):
         """Shorthand to stop firing."""
+    def color(self):
```
Comment (Leo-MathGuy) [inline pyndustri.pyi:350]: pyi stuff

File: pyndustric/compiler.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 from dataclasses import dataclass
 from pathlib import Path
 from typing import Callable, Union
+from string import hexdigits
```
Comment (Leo-MathGuy) [inline pyndustric/compiler.py:5]: Needed for hex check

**Comment by Leo-MathGuy** [comment]
oops misclicked

File: pyndustric/compiler.py (hunk 2 above)
Comment (Leo-MathGuy) [inline pyndustric/compiler.py:5]: Needed for hex check

File: pyndustric/compiler.py [hunk 3]
```diff
@@ -716,6 +717,11 @@ def emit_screen_syscall(self, node: ast.Call):
             else:
                 raise CompilerError(ERR_BAD_SYSCALL_ARGS, node)
 
+        elif method == "col":
```
Comment (Leo-MathGuy) [inline pyndustric/compiler.py:720]: Screen.col()

File: pyndustric/compiler.py [hunk 4]
```diff
@@ -1017,6 +1023,14 @@ def emit_control_syscall(self, node: ast.Call):
                 raise CompilerError(ERR_BAD_SYSCALL_ARGS, node)
 
             self.ins_append(f"control shoot {link} 0 0 0")
+        elif method == "color":
```
Comment (Leo-MathGuy) [inline pyndustric/compiler.py:1026]: .color()

File: pyndustric/compiler.py [hunk 5]
```diff
@@ -1120,6 +1134,11 @@ def as_value(self, node, output: str = None):
             elif isinstance(node.value, (int, float)):
                 return str(node.value)
             elif isinstance(node.value, str):
+                if (
```
Comment (Leo-MathGuy) [inline pyndustric/compiler.py:1137]: Hex check

**Comment by Lonami** [comment]
Thanks.

**Pull request merged**